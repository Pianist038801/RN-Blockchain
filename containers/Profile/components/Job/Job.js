import React from 'react'
import PropTypes from 'prop-types'
import { View, Text } from 'react-native'
import VerifiedIcon from '../VerifiedIcon'
import { getDates } from '../../utils/date.js'
import styles from '../../../../style.js'

const Job = (props) => {
  const { job } = props
  const dates = getDates(job.start_date, job.end_date)
  const textStyle = date => date === 'Present' ?
    [styles.profileEntityDate, styles.profilePresentText] :
    styles.profileEntityDate

  return (
    <View style={styles.profileEntityContainer}>
      <VerifiedIcon type="job" entity={job} />
      <View style={styles.profileEntityContent}>
        <Text style={styles.profileEntityPrimaryText}>{job.position}</Text>
        <Text>{job.company}</Text>
      </View>
      <View style={styles.profileEntityRight}>
        <Text style={textStyle(dates.start_date)}>{ dates.start_date }</Text>
        <Text style={textStyle(dates.end_date)}>- { dates.end_date }</Text>
      </View>
    </View>
  )
}

Job.propTypes = {
  job: PropTypes.shape({
    start_date: PropTypes.string.isRequired,
    end_date: PropTypes.string.isRequired,
    company: PropTypes.string.isRequired,
    position: PropTypes.string.isRequired
  })
}

export default Job
