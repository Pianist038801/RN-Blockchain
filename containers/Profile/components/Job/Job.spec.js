import React from 'react'
import renderer from 'react-test-renderer'
import Job from './index'

describe('containers/Profile/components/Job', () => {
  it('renders correctly', () => {
    const props = {
      job: {
        company: 'Applied Blockchain',
        start_date: '2017-11-04',
        end_date: '2018-01-24',
        position: 'developer',
        verified: true
      }
    }

    const tree = renderer.create(
      <Job {...props} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
