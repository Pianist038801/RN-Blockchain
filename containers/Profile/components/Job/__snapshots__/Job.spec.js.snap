// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Profile/components/Job renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#fff",
      "borderWidth": 1,
      "flex": 0.5,
      "flexDirection": "row",
      "marginBottom": 5,
      "padding": 22.5,
      "width": 600,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 75,
        "resizeMode": "contain",
        "width": 75,
      }
    }
  />
  <View
    style={
      Object {
        "marginLeft": 22.5,
        "width": 360,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#569cee",
          "fontWeight": "bold",
        }
      }
    >
      developer
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Applied Blockchain
    </Text>
  </View>
  <View
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "textAlign": "right",
        }
      }
    >
      2017
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "textAlign": "right",
        }
      }
    >
      - 
      2018
    </Text>
  </View>
</View>
`;
